aten/src/ATen/native/quantized/QTensor.cpp

```diff
@@ -343,6 +343,7 @@ std::tuple<Tensor, Tensor> choose_qparams_optimized(
   TORCH_CHECK(numel <= input_tensor.numel(), "numel ", numel,
       " greater than input_tensor.numel() ", input_tensor.numel());
   const float* input_row = input_tensor.const_data_ptr<float>();
+  TORCH_CHECK(input_row != nullptr, "input tensor is empty and has no data");
```

Comment:
Should be TORCH_CHECK_VALUE to raise ValueError. Should this be before the numel check?